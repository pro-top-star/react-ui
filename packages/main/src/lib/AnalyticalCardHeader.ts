import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
