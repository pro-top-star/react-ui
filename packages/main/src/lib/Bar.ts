import { Bar } from '../components/Bar';

export { Bar };
