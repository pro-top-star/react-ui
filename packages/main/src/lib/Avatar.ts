import { Avatar } from '../webComponents/Avatar';
import type { AvatarPropTypes } from '../webComponents/Avatar';

export { Avatar };
export type { AvatarPropTypes };
