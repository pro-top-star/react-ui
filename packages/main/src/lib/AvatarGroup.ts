import { AvatarGroup } from '../webComponents/AvatarGroup';
import type { AvatarGroupPropTypes } from '../webComponents/AvatarGroup';

export { AvatarGroup };
export type { AvatarGroupPropTypes };
