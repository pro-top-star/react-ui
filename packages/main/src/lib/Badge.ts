import { Badge } from '../webComponents/Badge';
import type { BadgePropTypes } from '../webComponents/Badge';

export { Badge };
export type { BadgePropTypes };
