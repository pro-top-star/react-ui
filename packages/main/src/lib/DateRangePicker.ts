import { DateRangePicker } from '../webComponents/DateRangePicker';
import type { DateRangePickerPropTypes } from '../webComponents/DateRangePicker';

export { DateRangePicker };
export type { DateRangePickerPropTypes };
