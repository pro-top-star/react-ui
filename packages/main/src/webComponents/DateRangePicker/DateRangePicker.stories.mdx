import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import { DateRangePicker } from '@ui5/webcomponents-react/lib/DateRangePicker';
import { CalendarType } from '@ui5/webcomponents-react/lib/CalendarType';
import '@ui5/webcomponents-localization/dist/features/calendar/Gregorian.js';
import '@ui5/webcomponents-localization/dist/features/calendar/Buddhist.js';
import '@ui5/webcomponents-localization/dist/features/calendar/Islamic.js';
import '@ui5/webcomponents-localization/dist/features/calendar/Japanese.js';
import '@ui5/webcomponents-localization/dist/features/calendar/Persian.js';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import { CSSProperties, Ref } from 'react';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';
import { DocsCommonProps } from '@shared/stories/DocsCommonProps';

<Meta
  title="Inputs / DateRangePicker"
  component={DateRangePicker}
  argTypes={{
    ...createSelectArgTypes({ primaryCalendarType: CalendarType, valueState: ValueState }),
    valueStateMessage: { control: { disable: true } },
    ...DocsCommonProps
  }}
  args={{
    primaryCalendarType: CalendarType.Gregorian,
    valueState: ValueState.None,
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    ref: null
  }}
/>

<DocsHeader />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return <DateRangePicker {...args} />;
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />
<div style={{ fontFamily: 'var(--sapFontFamily)', fontSize: 'var(--sapFontSize)', color: 'var(--sapTextColor)' }}>
  <h3>Usage</h3>
  The user can enter a date by: Using the calendar that opens in a popup or typing it in directly in the input field (not
  available for mobile devices).
</div>
