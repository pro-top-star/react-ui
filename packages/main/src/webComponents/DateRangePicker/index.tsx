import { CalendarType } from '@ui5/webcomponents-react/lib/CalendarType';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import { withWebComponent, WithWebComponentPropTypes } from '@ui5/webcomponents-react/lib/withWebComponent';
import '@ui5/webcomponents/dist/DateRangePicker';
import { FC, ReactNode } from 'react';

export interface DateRangePickerPropTypes extends Omit<WithWebComponentPropTypes, 'onChange' | 'onInput'> {
  /**
   * Determines the symbol which separates the dates.
   */
  delimiter?: string;
  /**
   * Determines whether the `DatePicker` is displayed as disabled.
   */
  disabled?: boolean;
  /**
   * Determines the format, displayed in the input field.
   */
  formatPattern?: string;
  /**
   * Defines the visibility of the week numbers column.
   *
   * **Note: **For calendars other than Gregorian, the week numbers are not displayed regardless of what is set.****
   */
  hideWeekNumbers?: boolean;
  /**
   * Determines the maximum date available for selection.
   */
  maxDate?: string;
  /**
   * Determines the minimum date available for selection.
   */
  minDate?: string;
  /**
   * Determines the name with which the `DatePicker` will be submitted in an HTML form.
   *
   * **Important:** For the `name` property to have effect, you must add the following import to your project: `import "@ui5/webcomponents/dist/features/InputElementsFormSupport.js";`
   *
   * **Note:** When set, a native `input` HTML element will be created inside the `DatePicker` so that it can be submitted as part of an HTML form. Do not use this property unless you need to submit a form.
   */
  name?: string;
  /**
   * Defines a short hint, intended to aid the user with data entry when the `DatePicker` has no value.
   *
   * **Note:** When no placeholder is set, the format pattern is displayed as a placeholder. Passing an empty string as the value of this property will make the `DatePicker` appear empty - without placeholder or format pattern.
   */
  placeholder?: string;
  /**
   * Determines the calendar type. The input value is formated according to the calendar type and the picker shows the months and years from the specified calendar.
   *
   * Available options are:
   *
   * *   `Gregorian`
   * *   `Islamic`
   * *   `Japanese`
   * *   `Buddhist`
   * *   `Persian`<br/><b>Note:</b> Calendar types other than Gregorian must be imported manually:<br /><code>import "@ui5/webcomponents-localization/dist/features/calendar/{primaryCalendarType}.js";</code>
   */
  primaryCalendarType?: CalendarType;
  /**
   * Determines whether the `DatePicker` is displayed as read-only.
   */
  readonly?: boolean;
  /**
   * Defines whether the `DatePicker` is required.
   */
  required?: boolean;
  /**
   * Defines a formatted date value.
   */
  value?: string;
  /**
   * Defines the value state of the `DatePicker`.
   *
   * Available options are:
   *
   * *   `None`
   * *   `Error`
   * *   `Warning`
   * *   `Success`
   * *   `Information`
   */
  valueState?: ValueState;
  /**
   * Defines the value state message that will be displayed as pop up under the `DatePicker`.
   *
   * **Note:** If not specified, a default text (in the respective language) will be displayed.
   * **Note:** The `valueStateMessage` would be displayed, when the `DatePicker` is in `Information`, `Warning` or `Error` value state.
   */
  valueStateMessage?: ReactNode;
  /**
   * Fired when the input operation has finished by pressing Enter or on focusout.
   */
  onChange?: (event: CustomEvent) => void;
  /**
   * Fired when the value of the `DatePicker` is changed at each key stroke.
   */
  onInput?: (event: CustomEvent) => void;
}

/**
 * The DateRangePicker enables the users to enter a localized date range using touch, mouse, keyboard input, or by selecting a date range in the calendar.
 *
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/DateRangePicker" target="_blank">UI5 Web Components Playground</a>
 */
const DateRangePicker: FC<DateRangePickerPropTypes> = withWebComponent<DateRangePickerPropTypes>(
  'ui5-daterange-picker',
  [
    'delimiter',
    'formatPattern',
    'maxDate',
    'minDate',
    'name',
    'placeholder',
    'primaryCalendarType',
    'value',
    'valueState'
  ],
  ['disabled', 'hideWeekNumbers', 'readonly', 'required'],
  ['valueStateMessage'],
  ['change', 'input']
);

DateRangePicker.displayName = 'DateRangePicker';

DateRangePicker.defaultProps = {
  delimiter: '-',
  disabled: false,
  hideWeekNumbers: false,
  placeholder: undefined,
  primaryCalendarType: CalendarType.Gregorian,
  readonly: false,
  required: false,
  valueState: ValueState.None
};

export { DateRangePicker };
