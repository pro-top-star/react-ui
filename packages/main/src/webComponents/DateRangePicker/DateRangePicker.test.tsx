import { DateRangePicker } from '@ui5/webcomponents-react/lib/DateRangePicker';
import { mount } from 'enzyme';
import React from 'react';

describe('DateRangePicker', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mount(<DateRangePicker />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
