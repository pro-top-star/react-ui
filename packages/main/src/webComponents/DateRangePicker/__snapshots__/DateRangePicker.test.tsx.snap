// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePicker Basic Test (generated) 1`] = `
<ui5-daterange-picker
  delimiter="-"
  primary-calendar-type="Gregorian"
  value-state="None"
/>
`;
